BaristaSwift/Common/Common/Base/Store.swift

```diff
@@ -89,95 +88,87 @@ public class Store<objectType: StoreProtocol> {
         }
         return results
     }
-
-    public func upsertEntries(jsonResponse: Any, completion: SyncCompletion = nil) {
-        let dataRows = (jsonResponse as! NSDictionary)["records"] as! [NSDictionary]
-        SalesforceSwiftLogger.log(type(of:self), level:.debug, message:"request:didLoadResponse: #records: \(dataRows.count)")
-        store.upsertEntries(dataRows, toSoup: objectType.objectName)
-        completion?(nil)
-    }
-    
-    public func upsertEntries<T:StoreProtocol>(record: T, completion: SyncCompletion = nil) {
-        store.upsertEntries([record.data], toSoup: T.objectName)
-        completion?(nil)
-    }
     
-    public func upsertNewEntries<T:StoreProtocol>(entry: T, completion: SyncCompletion = nil) {
-        var record: T = entry
+    public func locallyCreateEntry(entry: objectType) -> objectType {
```

Comment:
Better name for the functions.
Now we have three methods that only do local changes: locallyCreateEntry, locallyUpdateEntry and locallyDeleteEntry.
And also three methods that do the local changes followed by a sync: createEntry, updateEntry and deleteEntry.
Also it didn't make sense to have generic methods since the whole class is a generic type.
Finally the create and update methods are returning the object they created or updated (that simplifies calling code).

-